pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -59,14 +99,29 @@ t_ndarray   array_create(int32_t nd, int64_t *shape,
     }
     arr.buffer_size = arr.length * arr.type_size;
     arr.strides = malloc(nd * sizeof(int64_t));
-    for (int32_t i = 0; i < arr.nd; i++)
+    if (arr.order == order_c)
+    {
+        for (int32_t i = 0; i < arr.nd; i++)
+        {
+            arr.strides[i] = 1;
+            for (int32_t j = i + 1; j < arr.nd; j++)
+                arr.strides[i] *= arr.shape[j];
+        }
+    }
+    else if (arr.order == order_f)
     {
-        arr.strides[i] = 1;
-        for (int32_t j = i + 1; j < arr.nd; j++)
-            arr.strides[i] *= arr.shape[j];
+        for (int32_t i = 0; i < arr.nd; i++)
+        {
+            arr.strides[i] = 1;
+            for (int32_t j = 0; j < i; j++)
+                arr.strides[i] *= arr.shape[j];

```

Comment:
I think we need a discussion about how F ordered arrays work in C. Currently it is not clear to me if it is implemented in the same way as Fortran (implicitly by pyccel reversing the indexing when necessary), or in the same way as numpy (by changing the strides).
The last time I looked at this I got the impression that arguments were handled in a numpy-like way, but new arrays were handled in a fortran-like way. This needs checking. Unless this is homogenised here?

Reply:
Yes, I will write a doc about how I think it should be handled, so that we can discuss it.
In a nutshell, I thought it would make sense to copy Numpy's behaviour: *changing the strides, and the memory layout*

Reply:
I agree, but I think there are places in the code that try to mimic the Fortran handling so this may need modifying

Reply:
@EmilyBourne not sure if this is in discussion.

Reply:
I think I had a discussion with @ohachim about it. The discussion is not so much about which is better (I don't think there is a strong reason to prefer one over the other), but rather which is implemented and whether the convention is correctly followed throughout our code. The new doc should be invaluable when checking this as it will ensure that everyone is on the same page